admin/admin-service/src/main/java/io/mosip/admin/validator/DocTypeCodeValidator.java

```diff
@@ -0,0 +1,53 @@
+package io.mosip.admin.validator;
+
+import java.util.List;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import org.springframework.beans.factory.annotation.Autowired;
+
+import io.mosip.admin.bulkdataupload.repositories.DocumentTypeRepository;
+
+public class DocTypeCodeValidator implements ConstraintValidator<DocTypeCode, String>  {
+
+	private List<String> docTypeCode;
+
+	@Autowired
+	DocumentTypeRepository documentTypeRepository;
+	
+
+	@Override
+	public void initialize(DocTypeCode constraintAnnotation) {
+		if(documentTypeRepository == null){
+			/*         Note: An additional validation was getting triggered by doInvoke() method of 
+			 * 		 RepositoryListItemWriter class with documentTypeRepository equal to null
+			 *               which is not desired. This if clause is being used to escape that additional 
+			 *               validation step.
+			 */
+			return;
+		}
+		docTypeCode = documentTypeRepository.findAllByIsDeletedFalseOrIsDeletedIsNull();
```

Comment:
This will still not solve the problem. cached only on initialization, what happens to doc types which is created after initialization ?
Instead, the fix for the ticket should have been a catch clause where the exception is thrown and return a valid error-code.

Reply:
Removed that fix from this pr as it is not priority.